src/hooks/useSchemaValidator.ts

```diff
@@ -0,0 +1,218 @@
+import { useDatatype } from './useDatatype';
+import { useEffect, useMemo, useState } from 'react';
+import { useConfig } from '@openmrs/esm-framework';
+import { ConceptTrue, ConceptFalse } from '../constants';
+import { OHRIFormField, OHRIFormSchema } from '../api/types';
+
+export function useSchemaValidator(schema: OHRIFormSchema) {
+  //initializing state
+  const [errors, setErrors] = useState([]);
+  const [warnings, setWarnings] = useState([]);
+  const [fullArray, setFullArray] = useState<Array<OHRIFormField>>([]);
+  const [answersArray, setAnswersArray] = useState([]);
+  const [conceptSet, setConceptSet] = useState<Set<string>>(new Set());
+  const [answerConceptSet, setAnswerConceptSet] = useState<Set<string>>(new Set());
+  const { dataTypeToRenderingMap, conceptDataTypes } = useConfig();
+
+  const unresolvedConceptsFunc = (fullArray, filteredSetArray) => {
```

Comment:
The signature of this function can be improved, starting with the function name. Assuming it finds unresolved concepts, `findUnresolvedConcepts` is more descriptive. 

---
`fullArray`: It's not straightforward to tell the kind of objects this variable holds same applies to the other param

Reply:
Noted.